wikimedia/wikipedia-ios
# [HOLD MERGE] Update diff view for forthcoming Watchlist feature

**Phabricator:** https://phabricator.wikimedia.org/T335579 (partially)

### Notes
Updates diff view to new design in support of forthcoming Watchlist feature. I've retained the single revision view instead of removing it entirely mostly as a safety net, as there may be special cases or entry points where it still makes sense to see a single revision (like an article that only has a single revision). 

Ready for review - feel free to merge if you feel it's ready to be merged, or otherwise wait for forthcoming stacked PRs before merging into main.

Remaining work to come in future stacked PRs includes:
- Continuing design tweaks (some awaiting feedback from design in Phab task)
- Showing article image if available in header
- Improving next/previous revisions handling around boundaries of first revisions
- Updating header itself and header button design to not be full width
- Update Components framework menu button to support post initialization configuration (to handle our client pattern of lazy view configuration)
- Accessibility pass of the new work
- Hook up functionality of new menu buttons and header user buttons

### Test Steps
1. Open an article and view the edit history
2. Tap a single revision and confirm you see the updated design in the "compare" two revisions display
3. Confirm design update of main content, new toolbar icons and actions, and updated header
4. On a device with a large width (iPad), confirm new header author/edit details show side by side instead of stacked
